netty/src/main/java/io/grpc/netty/ProtocolNegotiators.java

```diff
@@ -95,6 +95,10 @@ public Handler newHandler(final GrpcHttp2ConnectionHandler handler) {
           public void handlerAdded(ChannelHandlerContext ctx) throws Exception {
             // Just replace this handler with the gRPC handler.
             ctx.pipeline().replace(this, null, handler);
+            // TODO(lukaszx0) not sure if this it the best/right place to do this
```

Comment:
Do it right before the replace(), since after this point `handler` may be receiving requests.